spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/RandomData.java

```diff
@@ -332,6 +340,20 @@ public Object map(Types.MapType map, Supplier<Object> keyResult, Supplier<Object
       return result;
     }
 
+    public VariantVal variant(Types.VariantType type) {
```

Comment:
Why is this conversion here in `RandomData`? It seems like it would be more appropriate in the test that uses it until we need it in more than one place.